src/main/scala/java/util/logging/Logger.scala

```diff
@@ -0,0 +1,322 @@
+package java.util.logging
+
+import scala.annotation.tailrec
+import scala.collection.mutable
+
+object Logger {
+
+  val GLOBAL_LOGGER_NAME: String =  "global"
+
+  // Not implemented, deprecated on JDK 1.8
+  //val global: Logger
+
+  private val defaultLogLevel: Level = Level.ALL
+
+  private val loggers: mutable.Map[String, Logger] = mutable.Map.empty
+
+  // Root is not visible to the outside but gives defaults
+  private[this] val rootLogger: Logger = {
+    val l = new Logger("", null)
+    l.setLevel(defaultLogLevel)
+    l.setUseParentHandlers(false)
+    l.setParent(null)
+    l
+  }
+
+  private[this] val globalLogger: Logger = {
+    val l = new Logger(GLOBAL_LOGGER_NAME, null)
+    l.setLevel(defaultLogLevel)
+    l.setUseParentHandlers(true)
+    l.setParent(rootLogger)
+    l
+  }
+
+  def getGlobal(): Logger = globalLogger
+
+  def getLogger(name: String): Logger = {
+    if (name == null)
+      throw new NullPointerException("Logger name cannot be null")
+
+    loggers.getOrElseUpdate(name, {
+      val l = new Logger(name, null)
```

Comment:
You could also put `newLogger` as a private method of the class add a `name` parameter and use it to implement `getAnonymousLogger`.